.changeset/tabaxi-fellas-inspire.md

```diff
@@ -0,0 +1,220 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+"__section": tree
+---
+TableSchema (alpha) APIs have been added
+
+A `TableSchema` utility has been added to Shared Tree for managing dynamic, tabular data.
+This new `TableSchema` namespace contains APIs for creating column, row, and table [node schema](https://fluidframework.com/docs/api/fluid-framework/treenodeschema-typealias).
+
+Note: these APIs require the use of [SchemaFactoryAlpha](https://fluidframework.com/docs/api/fluid-framework/schemafactoryalpha-class).
+
+> [!WARNING]
+> These APIs are in preview and are subject to change.
+> Until these APIs have stabilized, we do not recommend using them in production code.
+> We reserve the right to make breaking changes to these APIs, including their persisted data format.
+> Using these APIs in production code may result in data loss or corruption.
+
+#### Creating a table
+
+You can craft a table schema with `TableSchema.table`.
+Note that you will still be required to provide a schema for the cells that will appear in the table.
+
+```typescript
+class Table extends TableSchema.table({
+	schemaFactory,
+	cell: schemaFactory.string,
+}) {}
+
+const table = new Table({
+	columns: [{ id: "column-0" }],
+	rows: [{ id: "row-0", cells: { "column-0": "Hello world!" } }],
+});
+```
+
+#### Creating a table with custom column and row schema
+
+If you need to associate additional data with your rows or columns, you can customize your row and column schema via `TableSchema.column` and `TableSchema.row`.
+These schema can then be provided to `TableSchema.table`:
+
+```typescript
+const Cell = schemaFactory.string;
+
+class ColumnProps extends schemaFactory.object("TableColumnProps", {
+	/** The column label to display. */
+	label: schemaFactory.string,
+}) {}
+
+class Column extends TableSchema.column({
+	schemaFactory,
+	props: ColumnProps,
+}) {}
+
+class Row extends TableSchema.row({
+	schemaFactory,
+	cell: Cell,
+}) {}
+
+class Table extends TableSchema.table({
+	schemaFactory,
+	cell: Cell,
+	column: Column,
+	row: Row,
```

Comment:
Code like this makes me wonder if it would make sense to capitalize the properties which are passed schema so you could just do:

```
	Cell,
	Column,
	Row,
```

Reply:
Yeah, I've been wondering the same. Our SharedTree examples leverage `PascalCase` for derived schema classes _consistently_, yes? Or, at least, that is generally the guidance we use / recommend? If so, I'm good with such a rename.

Reply:
I try and recommend PascalCase for schema everywhere, but we don't often have schema as members of objects where this pattern would make sense. We do have a few though, and they don't follow this pattern for example the schema in TreeViewConfiguration is called schema not Schema.

Another example, SchemaFactory.string is lowercase.

I think I'm going to side against my own idea here since it would be really odd in cases where the schema are AlowedTypes unions. Also, I think we want to encourage people to use more semantic names than Cell, Column, and Row.

Reply:
I think the shorthand would be slick, but only applicable sometimes. I agree with Craig.